### PR TITLE
[docs] - Dagster Cloud native secrets

### PR DESCRIPTION
**Pull request by erinkcochran87**
### Summary & Motivation

This PR updates the environment variables and secrets documentation to include the new native secrets/env vars management feature for Dagster Cloud.

### CODE REVIEW

File: docs/content/guides/dagster/using-environment-variables-and-secrets.mdx
Comment (gibsondan) [inline docs/content/guides/dagster/using-environment-variables-and-secrets.mdx:24]: Any time the *`.env` file* is modified maybe

File: docs/content/guides/dagster/using-environment-variables-and-secrets.mdx
Comment (gibsondan) [inline docs/content/guides/dagster/using-environment-variables-and-secrets.mdx:19]: the .env file thing only works starting in 1.1 (not sure if we need to call that out or the docs versioning is enough)

Comment (gibsondan) [inline docs/content/guides/dagster/using-environment-variables-and-secrets.mdx:19]: Should we give an example of what a .env file looks like here?

File: docs/content/dagster-cloud/deployment/agents/customizing-configuration.mdx
Comment (erinkcochran87) [reply docs/content/dagster-cloud/deployment/agents/customizing-configuration.mdx:25]: I think this is a weird leftover from when I originally forked this branch. Gaaaaah.

File: docs/content/dagster-cloud/developing-testing/environment-variables-and-secrets.mdx
Comment (erinkcochran87) [reply docs/content/dagster-cloud/developing-testing/environment-variables-and-secrets.mdx:204]: Out of curiosity, would this still be a nit if each secret under the account-level key were encrypted with its own data key instead of all secrets being encrypted with the account-level key?

File: docs/content/dagster-cloud/developing-testing/environment-variables-and-secrets.mdx
Comment (erinkcochran87) [reply docs/content/dagster-cloud/developing-testing/environment-variables-and-secrets.mdx:232]: TBH it wasn't clear to me that this error was specific to that use case. Does this only come up then?

File: docs/content/dagster-cloud/developing-testing/environment-variables-and-secrets.mdx
Comment (erinkcochran87) [reply docs/content/dagster-cloud/developing-testing/environment-variables-and-secrets.mdx:290]: Good to know - based on that, this is what I understand the order of things to be - could you confirm?

1. Create env var in the UI.
2. Click **Save** to save/create the variable.
3. Re-deploy the branch / re-push the branch's PR
4. Branch picks up the new variable
5. 🎉

File: docs/content/dagster-cloud/developing-testing/environment-variables-and-secrets.mdx
Comment (gibsondan) [reply docs/content/dagster-cloud/developing-testing/environment-variables-and-secrets.mdx:232]: Yeah, since environment variables can be accessed anywhere within your code we don't have a lot of control in general over how a failure resulting in one being missing will manifest

File: docs/content/guides/dagster/using-environment-variables-and-secrets.mdx
Comment (erinkcochran87) [reply docs/content/guides/dagster/using-environment-variables-and-secrets.mdx:19]: It doesn't hurt to be explicit about it 👍

File: docs/content/guides/dagster/using-environment-variables-and-secrets.mdx
Comment (erinkcochran87) [reply docs/content/guides/dagster/using-environment-variables-and-secrets.mdx:21]: Is this a common use case? I'm not opposed to adding it now if it'll be quick to explain (specifically, that there's not much beyond what you said here) - I just don't want to hold up these docs if it's not a showstopper.

File: docs/content/guides/dagster/using-environment-variables-and-secrets.mdx
Comment (erinkcochran87) [reply docs/content/guides/dagster/using-environment-variables-and-secrets.mdx:19]: Is there a benefit to that?

File: docs/content/dagster-cloud/developing-testing/environment-variables-and-secrets.mdx
Comment (erinkcochran87) [reply docs/content/dagster-cloud/developing-testing/environment-variables-and-secrets.mdx:232]: Nope, nvm, got it from a comment further down. I'm just gonna take this out.

File: docs/content/dagster-cloud/developing-testing/environment-variables-and-secrets.mdx
Comment (erinkcochran87) [reply docs/content/dagster-cloud/developing-testing/environment-variables-and-secrets.mdx:113]: Originally my thinking was "You need to be able to modify the agent to do this" but that's not totally true - you just need to be able to modify `dagster_cloud.yaml` or `dagster.yaml`, correct?

File: docs/content/dagster-cloud/developing-testing/environment-variables-and-secrets.mdx
Comment (erinkcochran87) [reply docs/content/dagster-cloud/developing-testing/environment-variables-and-secrets.mdx:72]: I think the question here is do we still want to make this option an obvious choice in these docs. It feels much like the GH secrets method, in that it's more steps and more complicated and more likely to trip users up.

We **could** plunk the Hybrid env vars stuff in its own guide to de-emphasize it and then link to it from here. Thoughts? I feel like having it in here muddies the UI story and adds unnecessary complexity. Additionally, if we're saying the UI is the recommended way of doing things, I feel comfortable doing this.

**Comment by erinkcochran87** [comment]
@gibsondan Back to you - thanks for the detailed comments! This is what's still outstanding from the last round:


- [This bit about de-emphasizing agent config](https://github.com/dagster-io/dagster/pull/10486#discussion_r1023203214)
- [Can you confirm this is the correct order for env vars + Branch deps?](https://github.com/dagster-io/dagster/pull/10486#discussion_r1023270697)
- [Go/no go on running your own gRPC server](https://github.com/dagster-io/dagster/pull/10486#discussion_r1023273751)

File: docs/content/dagster-cloud/developing-testing/environment-variables-and-secrets.mdx
Comment (gibsondan) [reply docs/content/dagster-cloud/developing-testing/environment-variables-and-secrets.mdx:72]: I think there are two different things here:
- Platform-specific secrets like kubernetes secrets or aws secretsmanager - it seems important that we have clear instructions for people who don't want to send us their secret values, so I think we need to keep that part
- Hybrid env vars where you give us the actual key-value pairs in container_context (i'm fine leaving this out since it's not any better than the new UI)

File: docs/content/dagster-cloud/developing-testing/environment-variables-and-secrets.mdx
Comment (gibsondan) [reply docs/content/dagster-cloud/developing-testing/environment-variables-and-secrets.mdx:113]: Modifying dagster.yaml (should be called agent.yaml as per previous conversations) is basically the same as modifying the agent, but yeah modifying the data that dagster_cloud.yaml can be done in the github repo or from the dagster cloud UI on the code locations tab

File: docs/content/guides/dagster/using-environment-variables-and-secrets.mdx
Comment (gibsondan) [reply docs/content/guides/dagster/using-environment-variables-and-secrets.mdx]: That's about it but i think we can leave it out for now - i don't think 'running your own grpc server' and 'local dev' go together

File: docs/content/guides/dagster/using-environment-variables-and-secrets.mdx
Comment (gibsondan) [reply docs/content/guides/dagster/using-environment-variables-and-secrets.mdx]: I'm just not sure that it's a universally understood thing (the key=value format for example might not be obvious). I'd be happy to defer to whatever vercel does here though

File: docs/content/dagster-cloud/developing-testing/environment-variables-and-secrets.mdx
Comment (gibsondan) [reply docs/content/dagster-cloud/developing-testing/environment-variables-and-secrets.mdx:288]: That's right yeah

File: docs/content/guides/dagster/using-environment-variables-and-secrets.mdx
Comment (erinkcochran87) [reply docs/content/guides/dagster/using-environment-variables-and-secrets.mdx]: Oh, Vercel <3 I don't mind putting that in here to make it clear; I was just curious.

File: docs/content/dagster-cloud/developing-testing/environment-variables-and-secrets.mdx
Comment (gibsondan) [reply docs/content/dagster-cloud/developing-testing/environment-variables-and-secrets.mdx:72]: As per slack convo i think we can link to the hybrid agent docs here for now

**Review by gibsondan — APPROVED** [review]

**Comment by erinkcochran87** [comment]
@gibsondan This is back to you. I added instructions for configuring env vars for each agent type for full deployments and a placeholder for the exporting to `.env.` file feature. Once I can get screenshots of that from our UI, I'll add them in.

File: docs/content/dagster-cloud/developing-testing/environment-variables-and-secrets.mdx
Comment (gibsondan) [inline docs/content/dagster-cloud/developing-testing/environment-variables-and-secrets.mdx:411]: similar to ECS - I'd probably first recommend that somebody put a k8s secret here rather than env var key-value pairs. the nice thing about k8s secrets is that they stay more secret, and you can add and remove env vars without having to go back and do this again

File: docs/content/dagster-cloud/developing-testing/environment-variables-and-secrets.mdx
Comment (gibsondan) [inline docs/content/dagster-cloud/developing-testing/environment-variables-and-secrets.mdx:364]: another (possibly better?) way of doing this would be to use the secretsmanager integration here.

There is a secrets and secrets_tag field here (just like in the run launcher in OSS: https://docs.dagster.io/deployment/guides/aws#secrets-management-in-ecs) that works in the same way.

Frustratingly these fields are not currently in the ECS agent configuration reference

The reason I say better is that you don't have to put the value here, and if you use the tag you can add and remove them without needing to add them here. But that makes an already long section even longer.

**Review by gibsondan — COMMENTED** [review]
My big feedback on the hybrid agents stuff is that we now have an excellent step-by-step guide for the much less user friendly option, but the thing I would actually recommend a user do (go into the Code Locations tab or Github and add the thing to container_context) doesn't have a step by step guide yet - and the options that we would typically send a user to first for per-deployment stuff since they let you set secets without needing to put the actual values in your dagster.yaml (AWS secretsmanager for ECS and k8s secrets for K8s) aren't covered. Having some docs for this is probably better than the status quo where we have no docs though? Not sure. We could potentially land and then iterate on it to flesh it out more. Or we should separate out the hybrid agents stuff to a separate PR to ensure that the docs for the new UI can make it in for the release tomorrow

File: docs/content/dagster-cloud/developing-testing/environment-variables-and-secrets.mdx
Comment (erinkcochran87) [reply docs/content/dagster-cloud/developing-testing/environment-variables-and-secrets.mdx]: I'm gonna punt on this for a follow up PR, but it can definitely be added in

File: docs/content/dagster-cloud/developing-testing/environment-variables-and-secrets.mdx
Comment (erinkcochran87) [reply docs/content/dagster-cloud/developing-testing/environment-variables-and-secrets.mdx]: I'm gonna punt on this for a follow up PR, but it can definitely be added in

**Pull request merged**
